magic-nix-cache/src/main.rs

```diff
@@ -269,14 +270,35 @@ async fn main_cli() -> Result<()> {
             .as_bytes(),
         )
         .with_context(|| "Writing the post-build hook")?;
-        file.keep()
+        let path = file
+            .keep()
```

Comment:
do we need to keep it still if we copy to store and use it from there?

Reply:
related, probably don't need to clean it up if keep is removed

Reply:
It's deleted below if the `nix` invocation succeeds. We still use the original path if that fails for some reason.